exercises/clock/test/test_clock.c

```diff
@@ -16,406 +15,524 @@ void tearDown(void)
 {
 }
 
-// general routines that test expected results
-static void test_expected_value(const char *expected_clock_text, int hour,
-                                int minute)
+// test helper functions that test expected results
+static void test_expected_value(const char *expected_clock_text, clock_t clock)
 {
-   memset(actual_time_text, '\0', sizeof(actual_time_text));
-   clock(actual_time_text, hour, minute);
-   TEST_ASSERT_EQUAL_STRING(expected_clock_text, actual_time_text);
+   memset(actual_clock_text, '\0', sizeof(actual_clock_text));
+   clock_create(actual_clock_text, clock);
+   TEST_ASSERT_EQUAL_STRING(expected_clock_text, actual_clock_text);
 }
 
-static void test_time_add_expected_value(const char *expected_clock_text,
-                                         int hour, int minute,
-                                         int minute_offset)
+static void test_clock_add_expected_value(const char *expected_clock_text,
+                                          clock_t clock, int minute_offset)
 {
-   memset(actual_time_text, '\0', sizeof(actual_time_text));
-   clock(actual_time_text, hour, minute);
-   clock_add(actual_time_text, minute_offset);
+   memset(actual_clock_text, '\0', sizeof(actual_clock_text));
+   clock_create(actual_clock_text, clock);
+   clock_add(actual_clock_text, minute_offset);
+   TEST_ASSERT_EQUAL_STRING(expected_clock_text, actual_clock_text);
+}
 
-   TEST_ASSERT_EQUAL_STRING(expected_clock_text, actual_time_text);
+static void test_clock_subtract_expected_value(const char *expected_clock_text,
+                                               clock_t clock, int minute_offset)
+{
+   memset(actual_clock_text, '\0', sizeof(actual_clock_text));
+   clock_create(actual_clock_text, clock);
+   clock_subtract(actual_clock_text, minute_offset);
+   TEST_ASSERT_EQUAL_STRING(expected_clock_text, actual_clock_text);
 }
 
 // the tests
 static void test_on_the_hour(void)
 {
-   int hour = 8;
-   int minute = 0;
+   clock_t clock = { 8, 0 };
    char *expected_clock_text = "08:00";
```

Comment:
This can be written as

```c
clock_text_t expected_clock_text = "08:00";
```

Which will help us eliminate some additional type weirdness.